oscarlab/graphene
# [Pal/Linux-SGX] Replace SHA256 with HKDF-SHA256 in key derivations

## Description of the changes 

Pipe session shared key and process session shared key were previously generated using SHA256 as a Key Derivation Function HKDF). However, it is recommended to use HKDF-SHA256 as a KDF.

~~Should be rebased on top of #2380.~~

## How to test this PR? 

All tests must pass. You can use GDB to verify that the new HKDF-SHA256 works as expected.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/oscarlab/graphene/2410)
